### PR TITLE
19 try new architecture

### PR DESCRIPTION
**Pull request by DanielGoman**

### CODE REVIEW

File: src/model/dataset.py [hunk 1]
```diff
@@ -128,17 +118,13 @@ def __getitem__(self, idx: int):
...
-        image_labels_pair = self.image_path_labels_pairs[idx]
-        image = image_labels_pair['image']
-        image_labels = image_labels_pair['labels']
-
-        piece_type = image_labels['piece_type']
-        piece_color = image_labels['piece_color']
-        is_piece = image_labels['is_piece']
+        image_label_pair = self.image_path_labels_pairs[idx]
+        image = image_label_pair['image']
+        image_label = image_label_pair['label']
```
Comment (DanielGoman) [inline src/model/dataset.py:124]: Do the transform inside the creation of the dataset and not in the `__getitem__` method

File: src/model/evaluate.py [hunk 2]
```diff
@@ -21,57 +21,37 @@ def eval_model(model, loader: torch.utils.data.DataLoader, state: str, log: logg
...
-        color_counts = torch.zeros(num_piece_colors)
+        class_correct_hits = torch.zeros(num_classes)
+        class_counts = torch.zeros(num_classes)
 
-        for i, (image, type_label, color_label, is_piece) in enumerate(loader):
-            type_pred_probs, color_pred_probs = model(image)
-            type_pred = torch.argmax(type_pred_probs, axis=1)
-            color_pred = torch.argmax(color_pred_probs, axis=1)
+        for i, (image, label) in enumerate(loader):
+            class_probs = model(image.to(device))
```
Comment (DanielGoman) [inline src/model/evaluate.py:30]: What's the point of the exp here? This doesn't make it a probability. Softmax maybe may make more sense in this case, at least for clarity (although maybe it does the same)

File: src/model/evaluate.py [hunk 3]
```diff
@@ -21,57 +21,37 @@ def eval_model(model, loader: torch.utils.data.DataLoader, state: str, log: logg
...
+            image.cpu()
 
-            type_label = torch.argmax(type_label, axis=1)
-            color_label = torch.argmax(color_label, axis=1)
+            class_label = torch.argmax(label, axis=1)
 
-            type_correct_hits[type_label] += (type_pred == type_label).to(torch.int64)
-            labels_count_per_class = torch.bincount(type_label)
-            type_counts[labels_count_per_class.nonzero()] += labels_count_per_class[labels_count_per_class.nonzero()]
+            class_correct_hits[class_label] += (class_pred == class_label).to(torch.int64)
```
Comment (DanielGoman) [inline src/model/evaluate.py:36]: This only makes sense in the `batch_size=1` case. 
Otherwise, we need a count of correct classifications in the current batch

File: src/model/train.py [hunk 4]
```diff
@@ -83,39 +83,36 @@ def train(config: DictConfig) -> (str, torch.utils.data.DataLoader, torch.utils.
     model = PieceClassifier(in_channels=config.model_params.in_channels,
```
Comment (DanielGoman) [inline src/model/train.py:83]: Understand the point of 'is_minibatch' param

File: src/model/train.py [hunk 5]
```diff
@@ -83,39 +83,36 @@ def train(config: DictConfig) -> (str, torch.utils.data.DataLoader, torch.utils.
...
-            type_pred, color_pred = model(images.to(device))
-            images.detach().cpu()
-
-            loss = type_criterion(type_pred, type_labels.argmax(dim=1))
+            class_pred = model(images)
 
-            is_piece_idx = is_piece.nonzero()
-            if len(is_piece_idx) > 0:
-                loss += color_criterion(color_pred[is_piece_idx], color_labels[is_piece_idx].argmax(dim=1))
+            loss = criterion(class_pred, label)
```
Comment (DanielGoman) [inline src/model/train.py:108]: Doesn't CrossEntropyLoss expect logits?

File: src/model/evaluate.py [hunk 6]
```diff
@@ -21,57 +21,37 @@ def eval_model(model, loader: torch.utils.data.DataLoader, state: str, log: logg
...
+            labels_count_per_class = torch.bincount(class_label)
+            class_counts[labels_count_per_class.nonzero()] += labels_count_per_class[labels_count_per_class.nonzero()]
```
Comment (DanielGoman) [inline src/model/evaluate.py:38]: Give short explanation of what this does (Counts the number of appearances of each class in this dataloader)

File: src/model/train.py [hunk 7]
```diff
@@ -188,14 +183,14 @@ def get_subset_dataloader(dataset: torch.utils.data.Dataset, subset_ratio: float
     return subset_loader
 
 
-def plot_learning_curves(epoch_losses: List[float], epoch_train_accuracy: Dict[str, List[float]],
-                         epoch_val_accuracy: Dict[str, List[float]] = None, plots_path: str = ''):
+def plot_learning_curves(epoch_losses: List[float], epoch_train_accuracies: Dict[str, List[float]],
```
Comment (DanielGoman) [inline src/model/train.py:186]: Is thid needed, if we have TensorBoard logging?

**Review by DanielGoman — COMMENTED** [review]
Need some fleshing up on the code and some changes are needed.
Separate the notes into sub-issues if needed
